### ISSUE TITLE
Is it possible to add a column to show if APIs are CORS enabled?

### ISSUE THREAD

**Issue by tublitzed** [toddmotto/public-apis#303]
If this is something you wouldn't mind me doing, let me know and I'll open a PR for it. I figured I'd checkin here first to make sure this is something you'd be interested in before I do the work:)

I absolutely love this list and refer to it all the time, by the way. Awesome work putting this together.

**Comment by mgrigajtis** [comment]
+1 on this!

**Comment by mikestreety** [comment]
I'm not so familiar with CORS - mind explaining in laymans terms?

**Comment by mgrigajtis** [comment]
Able to make cross domain AJAX calls with client side JavaScript.

**Comment by maddenamy** [comment]
+1 on this!

**Comment by mikestreety** [comment]
Is there anyway we can automate/assume a yes/no without manually checking them all?

**Comment by Stroemgren** [comment]
You sure could, but you would need the API urls and not just the website urls as the list have at the moment.

In the case of having these urls, you could loop over them, making a request and checking the Access-Control-Allow-Origin header.

**Comment by jbrooksuk** [comment]
It's not fun, but we could add the column then retroactively update when we get time?

**Comment by alemesa1991** [comment]
That would be very helpful +1

**Comment by afabijan** [comment]
+1

**Comment by Tinee** [comment]
@tublitzed, did you pick it up? Otherwise, I could take a stab at it.

**Comment by mikestreety** [comment]
@Tinee I would say go go go.

Would suggest you make the default TBC or 🤷‍♀️ as a placeholder?

**Comment by davemachado** [comment]
@tublitzed @Tinee did one of you start working on this? Would hate to see you both duplicate work.

**Comment by Tinee** [comment]
Go for with @tublitzed :)

**Comment by GoToLoop** [comment]
I wonder whether a JSONP option should be included as an option as well.  :confused:

**Comment by wordythebyrd** [comment]
Came here to ask for this.

**Comment by mikestreety** [comment]
Anyone is more than welcome to pick this up :)

**Comment by torzuoliH** [comment]
Would be nice !

**Comment by davemachado** [comment]
Echoing @mikestreety 's comment that anyone is _encouraged_ to start working on this 🥇

**Comment by mikestreety** [comment]
Ok, looks like I need to tackle this. For now, I'll add an empty space or a something.

For people who don't know what CORS is, we should add a "how to check" or similar to the contributing guide.

So, how do you check if it's not obvious in the documentation of the API?

**Comment by davemachado** [comment]
Thanks for being willing to take this on @mikestreety. Maybe @tublitzed or @Tinee would still be willing to help out.

As far as my googling skills can tell me, the "easiest" way (besides conveniently specifying in the docs) would be to send a CORS request to an endpoint and see if it explodes or not. This could be done with postman or if you don't want to tediously hit them manually, maybe a small python/ruby script would help.

The problem here is for endpoints that require auth, hitting the endpoint might not be an option unless you feel like signing up for each and every service :) . To that, I'd offer trying to be sneaky and see if they have a demo endpoint you can hit.

**Comment by mikestreety** [comment]
@davemachado Quick Q, if I add this column now will it cock up your JSON parser?

**Comment by davemachado** [comment]
@mikestreety sorry just saw this now - it will certainly throw a wrench into the engine, but I will update accordingly

**Comment by sorxrob** [comment]
Hi guys! I made a simple site for this api list http://apis.robertsoriano.com

**Comment by kylecordes** [comment]
My only regret is that I have only one +1 to give :-)

Ideally someone comes up with all this stuff, it can be packaged in a form with a little widget that makes it easy to search/filter. Of course many of us are looking for something like the following to use in (For example) teaching classes, I would guess including @toddmotto ...

* free/public API have some kind
* works over HTTPS
* support for CORS so you can use it from a client-side app
* benign, "safer work" content that won't offend anyone

**Comment by davemachado** [comment]
Opened PR #589 to add a CORS column - please feel free to leave thoughts on the pull request 😄

**Comment by davemachado** [comment]
PR #589 merged. Closing issue 🎉
